1/tests/unit/ListItem.spec.js

```diff
@@ -0,0 +1,76 @@
+import { shallowMount } from '@vue/test-utils';
+import ListItem from '@/components/ListItem.vue';
+import { mount } from '@vue/test-utils';
+
+describe('ListItem.vue', () => {
+	it('render props.message when passed', () => {
+		const message = 'new todo';
+		const wrapper = shallowMount(ListItem, {
+			propsData: { message }
+		});
+		expect(wrapper.text()).toMatch(message);
+	});
+});
+
+describe('ListItem', () => {
+	describe('given a `todo`', () => {
+		const wrapper = mount(ListItem, {
+			name: 'item',
+			propsData: {
+				id: 1,
+				message: 'Foo'
+			}
+		});
+
+		
+		it('renders `item.message`', () => {
+			expect(wrapper.vm.message).toBe('Foo');
+		});
+		it("doesn't show input field", () => {
+			expect(wrapper.vm.onEdit).toBeFalsy();
+		});
+		it('has the expected html structure', () => {
+			expect(wrapper.element).toMatchSnapshot();
```

Comment:
While some developers like those snapshot tests, I'm not a huge fan of it. So what does `has the expected html structure` mean? I have to open another file and compare it in order to understand the test which is unnecessarily complex.